### PR TITLE
ICP-2152: LIFX - Devices are pairing double

### PR DESCRIPTION
**Pull request by munds**
This is a fix for parent app not uninstalling when there are no child devices present.

### CODE REVIEW

File: smartapps/smartthings/lifx-connect.src/lifx-connect.groovy [hunk 1]
```diff
@@ -269,9 +269,31 @@ def uninstalled() {
...
+	log.debug "Counting child devices"
+	state.attemptUninstall = true
+	runIn(11, attemptUninstall)
+}
+
+def attemptUninstall() {
+	log.debug "Will uninstall if no child devices"
+	if (!getChildDevices()) {
+		try {
+            log.debug "No children, performing uninstall"
```
Comment (larsfinander) [inline smartapps/smartthings/lifx-connect.src/lifx-connect.groovy:282]: whitespace

File: smartapps/smartthings/lifx-connect.src/lifx-connect.groovy [hunk 2]
```diff
@@ -269,9 +269,31 @@ def uninstalled() {
 	cirrus.unregisterServiceManager()
 }
 
+def countChildren() {
+	log.debug "Counting child devices"
+	state.attemptUninstall = true
+	runIn(11, attemptUninstall)
```
Comment (workingmonk) [inline smartapps/smartthings/lifx-connect.src/lifx-connect.groovy:275]: only do this if current count is <=1

**Comment by bflorian** [comment]
This used to happen automatically. Do we understand why that’s not happening anymore?

Also, is this problem unique to LIFX or is it happening to all service manager devices?

**Comment by workingmonk** [comment]
if this is a platform issue (which i guess it is), this is affecting other service managers too. this should be fixed in DM and not patched in here.

**Comment by munds** [comment]
Closing this PR because the root cause can be solved by a DM fix.

**Pull request closed without merging**